config/vlan.py

```diff
@@ -189,97 +231,159 @@ def vlan_member():
+            if (is_port and clicommon.interface_is_in_portchannel(portchannel_member_table, port)):
+                ctx.fail("{} is part of portchannel!".format(port))
+
+            # TODO: MISSING CONSTRAINT IN YANG MODEL
+            if (clicommon.interface_is_untagged_member(db.cfgdb, port) and untagged):
+                ctx.fail("{} is already untagged member!".format(port))
+            
+            # checking mode status of port if its access, trunk or routed
+            if is_port:
+                port_data = config_db.get_entry('PORT',port)
+            
+            # if not port then is a port channel
+            elif not is_port:
+                port_data = config_db.get_entry('PORTCHANNEL',port)
+
+            if "mode" not in port_data: 
+                ctx.fail("{} is in routed mode!\nUse switchport mode command to change port mode".format(port))
+            else:
+                existing_mode = port_data["mode"]
+            
+            if existing_mode == "routed":
+                ctx.fail("{} is in routed mode!\nUse switchport mode command to change port mode".format(port))
+
+            mode_type = "access" if untagged else "trunk"
+            if existing_mode == "access" and mode_type == "trunk":  # TODO: MISSING CONSTRAINT IN YANG MODEL
+                ctx.fail("{} is in access mode! Tagged Members cannot be added".format(port))
+            
+            elif existing_mode == mode_type or (existing_mode == "trunk" and mode_type == "access"):
+                pass
+            
+            # in case of exception in list last added member will be shown to user
         
-        portchannel_member_table = db.cfgdb.get_table('PORTCHANNEL_MEMBER')
-
-        if (is_port and clicommon.interface_is_in_portchannel(portchannel_member_table, port)): # TODO: MISSING CONSTRAINT IN YANG MODEL
-            ctx.fail("{} is part of portchannel!".format(port))
-
-        if (clicommon.interface_is_untagged_member(db.cfgdb, port) and untagged): # TODO: MISSING CONSTRAINT IN YANG MODEL
-            ctx.fail("{} is already untagged member!".format(port))
-
-    try:
-        config_db.set_entry('VLAN_MEMBER', (vlan, port), {'tagging_mode': "untagged" if untagged else "tagged" })
-    except ValueError:
-        ctx.fail("{} invalid or does not exist, or {} invalid or does not exist".format(vlan, port))
+            try:
+                config_db.set_entry('VLAN_MEMBER', (vlan, port), {'tagging_mode': "untagged" if untagged else "tagged" })
+            except ValueError:
+                ctx.fail("{} invalid or does not exist, or {} invalid or does not exist".format(vlan, port))
 
 @vlan_member.command('del')
-@click.argument('vid', metavar='<vid>', required=True, type=int)
+@click.argument('vid', metavar='<vid>', required=True)
 @click.argument('port', metavar='<port>', required=True)
+@click.option('-m', '--multiple', is_flag=True, help="Add Multiple Vlan(s) in Range or in Comma separated list")
```

Comment:
same as above

Reply:
Please note that --multiple flag is required as we did not add our new functionality to the existing command to keep intact the existing VLAN behavior/command.